### PR TITLE
feat(core): add method to update consentHub

### PR DESCRIPTION
**Pull request by Johaney-s**
* added method to update consentHub
* adding facilities this way is not possible

### CODE REVIEW

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
It looks good, I have just a few suggestions.

**Review by cuadradek — CHANGES_REQUESTED** [review]

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntry.java [hunk 1]
```diff
@@ -117,4 +117,20 @@ public ConsentHub getConsentHubByFacility(PerunSession sess, int facilityId) thr
 		return consentsManagerBl.getConsentHubByFacility(sess, facilityId);
 	}
 
+	@Override
+	public ConsentHub updateConsentHub(PerunSession sess, ConsentHub consentHub) throws ConsentHubNotExistsException, PrivilegeException {
+		Utils.checkPerunSession(sess);
+
+		getConsentsManagerBl().getConsentHubById(sess, consentHub.getId());
```
[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntry.java]: This supposed to be a check that consentHub we're updating exists, but I agree `checkConsentHubExists` will be more clear.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java [hunk 2]
```diff
@@ -174,6 +174,23 @@ public ConsentHub createConsentHub(PerunSession sess, ConsentHub consentHub) {
 		}
 	}
 
+	public void updateConsentHub(PerunSession sess, ConsentHub consentHub) {
+		try {
+			// if name not set, use facility name
+			if (consentHub.getName() == null) {
```
[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java]: As @cuadradek noticed, it should not be allowed for updating hub, but added check for null or empty name to entry layer.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntry.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntry.java]: good point, replaced

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java [hunk 3]
```diff
@@ -174,6 +174,23 @@ public ConsentHub createConsentHub(PerunSession sess, ConsentHub consentHub) {
...
+				consentHub.setName(consentHub.getFacilities().get(0).getName());
```
[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java]: absolutely, added the check to the entry layer

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java [hunk 4]
```diff
@@ -174,6 +174,23 @@ public ConsentHub createConsentHub(PerunSession sess, ConsentHub consentHub) {
...
+			}
+
+			jdbc.update("update consent_hubs set name=?, enforce_consents=?, modified_by=?, modified_by_uid=?," +
+					" modified_at=" + Compatibility.getSysdate() + " where id=?", consentHub.getName(), consentHub.isEnforceConsents(),
+					sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getUserId(), consentHub.getId());
```
[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java:184]: added with test for that

**Review by HejdaJakub — APPROVED** [review]

**Review by cuadradek — APPROVED** [review]

**Pull request merged**